src/main/scala/mesosphere/marathon/state/Container.scala

```diff
@@ -47,7 +47,14 @@ object Container {
         name: Option[String] = None,
         labels: Map[String, String] = Map.empty[String, String]) {
 
-      require(protocol == "tcp" || protocol == "udp", "protocol can only be 'tcp' or 'udp'")
+      require(
```

Comment:
The require statements are introduced, as we had weak validation.
But I agree. We should remove the require statements in a separate step.